appservice/src/SiteWrapper.ts

```diff
@@ -529,32 +521,86 @@ export class SiteWrapper {
+                    permanentId = deployment.id;
+                }
             }
 
-            if (deployment.progress) {
-                this.log(outputChannel, deployment.progress);
+            if (deployment === undefined || !deployment.id) {
+                throw new Error(localize('failedToFindDeployment', 'Failed to get status of deployment.'));
             }
 
-            await new Promise((resolve: () => void): void => { setTimeout(resolve, pollingInterval); });
-            deployment = await kuduClient.deployment.getResult(deploymentId);
-        }
-        if (previousDeploymentId !== undefined && deployment.id === previousDeploymentId) {
-            throw new Error(localize('unsuccessfulDeployment', 'The deployment was unsuccessful. If this is a Local Git deployment, this could be a duplicate deployment.'));
+            let logEntries: LogEntry[] = [];
+            try {
+                logEntries = <LogEntry[]>await kuduClient.deployment.getLogEntry(deployment.id);
+            } catch (error) {
+                // Swallow 404 errors for temp deployments
+                // (it's expected that the temp logs don't exist once the deployment switches to permanent)
+                // tslint:disable-next-line:no-unsafe-any
+                if (!deployment.isTemp || !error || error.statusCode !== 404) {
+                    throw error;
+                }
+            }
+
+            const newLogEntries: LogEntry[] = logEntries.filter((newEntry: LogEntry) => !alreadyDisplayedLogs.some((oldId: string) => newEntry.id === oldId));
+            if (newLogEntries.length === 0) {
+                if (Date.now() > nextTimeToDisplayWaitingLog) {
+                    this.log(outputChannel, localize('waitingForComand', 'Waiting for long running command to finish...'));
+                    nextTimeToDisplayWaitingLog = Date.now() + 60 * 1000;
+                }
+            } else {
+                for (const newEntry of newLogEntries) {
+                    if (newEntry.id) {
+                        alreadyDisplayedLogs.push(newEntry.id);
+                        if (newEntry.message) {
+                            this.log(outputChannel, newEntry.message, newEntry.logTime);
+                        }
+
+                        if (newEntry.detailsUrl) {
+                            const entryDetails: LogEntry[] = await kuduClient.deployment.getLogEntryDetails(deployment.id, newEntry.id);
+                            for (const entryDetail of entryDetails) {
+                                if (entryDetail.message) {
+                                    this.log(outputChannel, entryDetail.message, entryDetail.logTime);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (deployment.complete) {
+                if (deployment.isTemp) {
```

Comment:
complete means failure for temp deployments?

Reply:
Yep. Every deployment transitions from "temp" to "permanent". If it is set as "complete" during the "temp" phase, then it failed